swap/src/cli/event_loop.rs

```diff
@@ -299,98 +301,147 @@ impl EventLoop {
+                    self.inflight_cooperative_xmr_redeem_requests.insert(outbound_request_id, responder);
+
+                    tracing::trace!(
+                        %peer_id,
+                        %swap_id,
+                        %outbound_request_id,
+                        "Dispatching outgoing cooperative xmr redeem request"
+                    );
                 },
 
-                // We use `self.is_connected_to_alice` as a guard to "buffer" requests until we are connected.
+                // We use `self.swarm.is_connected` as a guard to "buffer" requests until we are connected.
                 // because the protocol does not dial Alice itself
                 // (unlike request-response above)
-                Some((swap, responder)) = self.execution_setup_requests.next().fuse(), if self.is_connected_to_alice() => {
-                    self.swarm.behaviour_mut().swap_setup.start(self.alice_peer_id, swap).await;
-                    self.inflight_swap_setup = Some(responder);
+                Some(((alice_peer_id, swap), responder)) = self.execution_setup_requests.next().fuse() => {
+                    let swap_id = swap.swap_id.clone();
+
+                    tracing::trace!(
+                        %alice_peer_id,
+                        "Dispatching outgoing execution setup request"
+                    );
+
+                    self.swarm.behaviour_mut().swap_setup.start(alice_peer_id, swap).await;
+                    self.inflight_swap_setup.insert((alice_peer_id, swap_id), responder);
                 },
 
                 // Send an acknowledgement to Alice once the EventLoopHandle has processed a received transfer proof
-                // We use `self.is_connected_to_alice` as a guard to "buffer" requests until we are connected.
+                // We use `self.swarm.is_connected` as a guard to "buffer" requests until we are connected.
                 //
                 // Why do we do this here but not for the other request-response channels?
                 // This is the only request, we don't have a retry mechanism for. We lazily send this.
-                Some(response_channel) = &mut self.pending_transfer_proof, if self.is_connected_to_alice() => {
-                    if self.swarm.behaviour_mut().transfer_proof.send_response(response_channel, ()).is_err() {
-                        tracing::warn!("Failed to send acknowledgment to Alice that we have received the transfer proof");
-                    } else {
-                        tracing::info!("Sent acknowledgment to Alice that we have received the transfer proof");
-                        self.pending_transfer_proof = OptionFuture::from(None);
+                Some((swap_id, response_channel)) = self.pending_transfer_proof_acks.next() => {
+                    if let Some(peer_id) = self.swap_peer_id(&swap_id) {
+                        if self.swarm.is_connected(&peer_id) {
```

Comment:
Not sure if it makes sense at all to buffer here because I don't think responding on a channel which has been dropped works even if a new connections has been established